elastic/beats
# Fix event mapping implementation for statsd module

- Bug

## Proposed commit message

The root cause of the problem is the implementation of the event mapping method.

The data periodically sent by Airflow is not limited to a single DAG; it can encompass information from all currently running DAGs. This collective data is gathered into a group and subsequently processed to generate an event. The issue was traced back to a for loop used to iterate through all the key-value pairs within this group to create a single event. Because the data originates from multiple DAGs, creating a single event resulted in the data from one DAG overwriting the data from another. This led to the chaotic mixing of metrics, causing the problem.

The new implementation matches the approach used in the statsd_exporter. Now, each key-value pair and its associated attributes are treated as individual events. As a result, each key-value pair is processed to create distinct events, with only the related fields present in each event. This modification effectively resolves the issue and offers greater flexibility when using controls and filters to generate visualizations for associated data.

## Checklist


## How to test this PR locally

1. Enable the statsd module.
2. When testing any module, such as airflow, configure the statsd mapping.
3. Verify the data and mappings in Elasticsearch.

## Related issues


- Closes https://github.com/elastic/sdh-beats/issues/3932
## Screenshots
